ggshield/iac/iac_scan_models.py

```diff
@@ -32,24 +38,53 @@ class IaCDiffScanResult(Base):
 )
 
 
-def mock_api_iac_diff_scan(
-    client: Any,
-    reference: bytes,
-    current: bytes,
-    scan_parameters: IaCScanParameters,
-    extra_headers: Optional[Dict[str, str]] = None,
-) -> Union[Detail, IaCDiffScanResult]:
-    scan = client.iac_directory_scan(
-        Path("."),
-        [],
-        scan_parameters,
-        extra_headers,
-    )
-    result = IaCDiffScanResult(
-        scan.id,
-        scan.type,
-        scan.iac_engine_version,
-        entities_with_incidents=IaCDiffScanEntities([], [], []),
-    )
-    result.status_code = 200
-    return result
+class MockClient(GGClient):
```

Comment:
@agateau-gg my idea here is to subclass the existing client until we are sure about the implementation and then to migrate into pygitguardian.
In this MR, the "real" routes to GIM are not implemented.